File: mobile/API/SupportCategories.swift
import Foundation

// MARK: - Support Categories Enum для iOS
public enum SupportCategory: String, CaseIterable, Codable {
    // Основные категории
    case general = "general"
    case security = "security"
    case family = "family"
    case settings = "settings"
    case help = "help"
    
    // Технические категории
    case technical = "technical"
    case installation = "installation"
    case configuration = "configuration"
    case troubleshooting = "troubleshooting"
    case performance = "performance"
    
    // Безопасность
    case parentalControl = "parental_control"
    case childProtection = "child_protection"
    case elderlyProtection = "elderly_protection"
    case deviceSecurity = "device_security"
    case networkSecurity = "network_security"
    case dataProtection = "data_protection"
    
    // VPN и сеть
    case vpn = "vpn"
    case network = "network"
    case connectivity = "connectivity"
    case firewall = "firewall"
    
    // Платежи и подписки
    case payments = "payments"
    case subscription = "subscription"
    case billing = "billing"
    case refunds = "refunds"
    
    // Кризисные ситуации
    case crisis = "crisis"
    case emergency = "emergency"
    case psychological = "psychological"
    case childSafety = "child_safety"
    case cyberbullying = "cyberbullying"
    
    // Дополнительные
    case feedback = "feedback"
    case suggestions = "suggestions"
    case complaints = "complaints"
    case other = "other"
    
    // MARK: - Computed Properties
    public var displayName: String {
        switch self {
        case .general:
            return "Общие вопросы"
        case .security:
            return "Безопасность"
        case .family:
            return "Семья"
        case .settings:
            return "Настройки"
        case .help:
            return "Помощь"
        case .technical:
            return "Техническая поддержка"
        case .installation:
            return "Установка"
        case .configuration:
            return "Конфигурация"
        case .troubleshooting:
            return "Устранение неполадок"
        case .performance:
            return "Производительность"
        case .parentalControl:
            return "Родительский контроль"
        case .childProtection:
            return "Защита детей"
        case .elderlyProtection:
            return "Защита пожилых"
        case .deviceSecurity:
            return "Безопасность устройства"
        case .networkSecurity:
            return "Сетевая безопасность"
        case .dataProtection:
            return "Защита данных"
        case .vpn:
            return "VPN"
        case .network:
            return "Сеть"
        case .connectivity:
            return "Подключение"
        case .firewall:
            return "Файрвол"
        case .payments:
            return "Платежи"
        case .subscription:
            return "Подписка"
        case .billing:
            return "Биллинг"
        case .refunds:
            return "Возвраты"
        case .crisis:
            return "Кризисная ситуация"
        case .emergency:
            return "Экстренная помощь"
        case .psychological:
            return "Психологическая поддержка"
        case .childSafety:
            return "Безопасность детей"
        case .cyberbullying:
            return "Кибербуллинг"
        case .feedback:
            return "Обратная связь"
        case .suggestions:
            return "Предложения"
        case .complaints:
            return "Жалобы"
        case .other:
            return "Другое"
        }
    }
    
    public var description: String {
        switch self {
        case .general:
            return "Общие вопросы и информация"
        case .security:
            return "Вопросы безопасности и защиты"
        case .family:
            return "Семейные настройки и профили"
        case .settings:
            return "Настройки приложения"
        case .help:
            return "Справка и инструкции"
        case .technical:
            return "Технические проблемы и решения"
        case .installation:
            return "Установка и обновления"
        case .configuration:
            return "Настройка параметров"
        case .troubleshooting:
            return "Диагностика и исправление ошибок"
        case .performance:
            return "Оптимизация производительности"
        case .parentalControl:
            return "Настройка родительского контроля"
        case .childProtection:
            return "Защита детей в интернете"
        case .elderlyProtection:
            return "Защита пожилых пользователей"
        case .deviceSecurity:
            return "Безопасность мобильного устройства"
        case .networkSecurity:
            return "Защита сетевого соединения"
        case .dataProtection:
            return "Защита персональных данных"
        case .vpn:
            return "Настройка и использование VPN"
        case .network:
            return "Сетевые подключения"
        case .connectivity:
            return "Проблемы с подключением"
        case .firewall:
            return "Настройка файрвола"
        case .payments:
            return "Платежи и транзакции"
        case .subscription:
            return "Управление подпиской"
        case .billing:
            return "Счета и оплата"
        case .refunds:
            return "Возврат средств"
        case .crisis:
            return "Кризисные ситуации и экстренная помощь"
        case .emergency:
            return "Экстренная поддержка"
        case .psychological:
            return "Психологическая помощь"
        case .childSafety:
            return "Экстренная помощь по безопасности детей"
        case .cyberbullying:
            return "Помощь при кибербуллинге"
        case .feedback:
            return "Отзывы и предложения"
        case .suggestions:
            return "Предложения по улучшению"
        case .complaints:
            return "Жалобы и проблемы"
        case .other:
            return "Другие вопросы"
        }
    }
    
    public var priority: SupportPriority {
        switch self {
        case .crisis, .emergency, .childSafety, .cyberbullying:
            return .critical
        case .psychological, .technical, .troubleshooting:
            return .high
        case .security, .parentalControl, .childProtection, .elderlyProtection:
            return .high
        case .vpn, .network, .connectivity, .deviceSecurity:
            return .medium
        case .payments, .subscription, .billing:
            return .medium
        case .settings, .configuration, .performance:
            return .medium
        case .general, .help, .feedback, .suggestions:
            return .low
        case .other, .complaints:
            return .low
        }
    }
    
    public var icon: String {
        switch self {
        case .general:
            return "questionmark.circle"
        case .security:
            return "shield.checkered"
        case .family:
            return "person.3"
        case .settings:
            return "gear"
        case .help:
            return "questionmark.circle.fill"
        case .technical:
            return "wrench.and.screwdriver"
        case .installation:
            return "arrow.down.circle"
        case .configuration:
            return "slider.horizontal.3"
        case .troubleshooting:
            return "stethoscope"
        case .performance:
            return "speedometer"
        case .parentalControl:
            return "person.crop.circle.badge.checkmark"
        case .childProtection:
            return "figure.child"
        case .elderlyProtection:
            return "figure.walk"
        case .deviceSecurity:
            return "iphone"
        case .networkSecurity:
            return "network"
        case .dataProtection:
            return "lock.shield"
        case .vpn:
            return "network"
        case .network:
            return "wifi"
        case .connectivity:
            return "antenna.radiowaves.left.and.right"
        case .firewall:
            return "shield.lefthalf.filled"
        case .payments:
            return "creditcard"
        case .subscription:
            return "repeat"
        case .billing:
            return "doc.text"
        case .refunds:
            return "arrow.uturn.backward"
        case .crisis:
            return "exclamationmark.triangle.fill"
        case .emergency:
            return "phone.fill"
        case .psychological:
            return "heart.fill"
        case .childSafety:
            return "figure.child.circle.fill"
        case .cyberbullying:
            return "person.crop.circle.badge.exclamationmark"
        case .feedback:
            return "bubble.left.and.bubble.right"
        case .suggestions:
            return "lightbulb"
        case .complaints:
            return "exclamationmark.bubble"
        case .other:
            return "ellipsis.circle"
        }
    }
    
    public var color: String {
        switch self {
        case .crisis, .emergency, .childSafety, .cyberbullying:
            return "error"
        case .psychological, .technical, .troubleshooting:
            return "warning"
        case .security, .parentalControl, .childProtection, .elderlyProtection:
            return "accent"
        case .vpn, .network, .connectivity, .deviceSecurity:
            return "primary"
        case .payments, .subscription, .billing:
            return "secondary"
        case .settings, .configuration, .performance:
            return "info"
        case .general, .help, .feedback, .suggestions:
            return "success"
        case .other, .complaints:
            return "text"
        }
    }
}

// MARK: - Support Priority Enum
public enum SupportPriority: String, CaseIterable, Codable {
    case low = "low"
    case medium = "medium"
    case high = "high"
    case critical = "critical"
    
    public var displayName: String {
        switch self {
        case .low:
            return "Низкий"
        case .medium:
            return "Средний"
        case .high:
            return "Высокий"
        case .critical:
            return "Критический"
        }
    }
    
    public var description: String {
        switch self {
        case .low:
            return "Обычный вопрос, не требует срочного решения"
        case .medium:
            return "Важный вопрос, требует внимания"
        case .high:
            return "Срочный вопрос, требует быстрого решения"
        case .critical:
            return "Критическая ситуация, требует немедленного решения"
        }
    }
    
    public var responseTime: TimeInterval {
        switch self {
        case .low:
            return 24 * 60 * 60 // 24 часа
        case .medium:
            return 4 * 60 * 60 // 4 часа
        case .high:
            return 1 * 60 * 60 // 1 час
        case .critical:
            return 15 * 60 // 15 минут
        }
    }
    
    public var color: String {
        switch self {
        case .low:
            return "success"
        case .medium:
            return "info"
        case .high:
            return "warning"
        case .critical:
            return "error"
        }
    }
}

// MARK: - Crisis Type Enum
public enum CrisisType: String, CaseIterable, Codable {
    case childSafety = "child_safety"
    case cyberbullying = "cyberbullying"
    case psychological = "psychological"
    case emergency = "emergency"
    case security = "security"
    case other = "other"
    
    public var displayName: String {
        switch self {
        case .childSafety:
            return "Безопасность детей"
        case .cyberbullying:
            return "Кибербуллинг"
        case .psychological:
            return "Психологическая помощь"
        case .emergency:
            return "Экстренная ситуация"
        case .security:
            return "Безопасность"
        case .other:
            return "Другое"
        }
    }
    
    public var description: String {
        switch self {
        case .childSafety:
            return "Критическая ситуация с безопасностью ребенка"
        case .cyberbullying:
            return "Случай кибербуллинга или травли"
        case .psychological:
            return "Психологическая кризисная ситуация"
        case .emergency:
            return "Экстренная ситуация, требующая немедленного вмешательства"
        case .security:
            return "Критическая угроза безопасности"
        case .other:
            return "Другая кризисная ситуация"
        }
    }
    
    public var urgencyLevel: Int {
        switch self {
        case .childSafety, .emergency:
            return 1
        case .cyberbullying, .security:
            return 2
        case .psychological:
            return 3
        case .other:
            return 4
        }
    }
}

File: mobile/API/SupportCategories.kt
package com.aladdin.mobile.api

import kotlinx.serialization.Serializable

// MARK: - Support Categories Enum для Android
@Serializable
enum class SupportCategory(val value: String) {
    // Основные категории
    GENERAL("general"),
    SECURITY("security"),
    FAMILY("family"),
    SETTINGS("settings"),
    HELP("help"),
    
    // Технические категории
    TECHNICAL("technical"),
    INSTALLATION("installation"),
    CONFIGURATION("configuration"),
    TROUBLESHOOTING("troubleshooting"),
    PERFORMANCE("performance"),
    
    // Безопасность
    PARENTAL_CONTROL("parental_control"),
    CHILD_PROTECTION("child_protection"),
    ELDERLY_PROTECTION("elderly_protection"),
    DEVICE_SECURITY("device_security"),
    NETWORK_SECURITY("network_security"),
    DATA_PROTECTION("data_protection"),
    
    // VPN и сеть
    VPN("vpn"),
    NETWORK("network"),
    CONNECTIVITY("connectivity"),
    FIREWALL("firewall"),
    
    // Платежи и подписки
    PAYMENTS("payments"),
    SUBSCRIPTION("subscription"),
    BILLING("billing"),
    REFUNDS("refunds"),
    
    // Кризисные ситуации
    CRISIS("crisis"),
    EMERGENCY("emergency"),
    PSYCHOLOGICAL("psychological"),
    CHILD_SAFETY("child_safety"),
    CYBERBULLYING("cyberbullying"),
    
    // Дополнительные
    FEEDBACK("feedback"),
    SUGGESTIONS("suggestions"),
    COMPLAINTS("complaints"),
    OTHER("other");
    
    // MARK: - Computed Properties
    val displayName: String
        get() = when (this) {
            GENERAL -> "Общие вопросы"
            SECURITY -> "Безопасность"
            FAMILY -> "Семья"
            SETTINGS -> "Настройки"
            HELP -> "Помощь"
            TECHNICAL -> "Техническая поддержка"
            INSTALLATION -> "Установка"
            CONFIGURATION -> "Конфигурация"
            TROUBLESHOOTING -> "Устранение неполадок"
            PERFORMANCE -> "Производительность"
            PARENTAL_CONTROL -> "Родительский контроль"
            CHILD_PROTECTION -> "Защита детей"
            ELDERLY_PROTECTION -> "Защита пожилых"
            DEVICE_SECURITY -> "Безопасность устройства"
            NETWORK_SECURITY -> "Сетевая безопасность"
            DATA_PROTECTION -> "Защита данных"
            VPN -> "VPN"
            NETWORK -> "Сеть"
            CONNECTIVITY -> "Подключение"
            FIREWALL -> "Файрвол"
            PAYMENTS -> "Платежи"
            SUBSCRIPTION -> "Подписка"
            BILLING -> "Биллинг"
            REFUNDS -> "Возвраты"
            CRISIS -> "Кризисная ситуация"
            EMERGENCY -> "Экстренная помощь"
            PSYCHOLOGICAL -> "Психологическая поддержка"
            CHILD_SAFETY -> "Безопасность детей"
            CYBERBULLYING -> "Кибербуллинг"
            FEEDBACK -> "Обратная связь"
            SUGGESTIONS -> "Предложения"
            COMPLAINTS -> "Жалобы"
            OTHER -> "Другое"
        }
    
    val description: String
        get() = when (this) {
            GENERAL -> "Общие вопросы и информация"
            SECURITY -> "Вопросы безопасности и защиты"
            FAMILY -> "Семейные настройки и профили"
            SETTINGS -> "Настройки приложения"
            HELP -> "Справка и инструкции"
            TECHNICAL -> "Технические проблемы и решения"
            INSTALLATION -> "Установка и обновления"
            CONFIGURATION -> "Настройка параметров"
            TROUBLESHOOTING -> "Диагностика и исправление ошибок"
            PERFORMANCE -> "Оптимизация производительности"
            PARENTAL_CONTROL -> "Настройка родительского контроля"
            CHILD_PROTECTION -> "Защита детей в интернете"
            ELDERLY_PROTECTION -> "Защита пожилых пользователей"
            DEVICE_SECURITY -> "Безопасность мобильного устройства"
            NETWORK_SECURITY -> "Защита сетевого соединения"
            DATA_PROTECTION -> "Защита персональных данных"
            VPN -> "Настройка и использование VPN"
            NETWORK -> "Сетевые подключения"
            CONNECTIVITY -> "Проблемы с подключением"
            FIREWALL -> "Настройка файрвола"
            PAYMENTS -> "Платежи и транзакции"
            SUBSCRIPTION -> "Управление подпиской"
            BILLING -> "Счета и оплата"
            REFUNDS -> "Возврат средств"
            CRISIS -> "Кризисные ситуации и экстренная помощь"
            EMERGENCY -> "Экстренная поддержка"
            PSYCHOLOGICAL -> "Психологическая помощь"
            CHILD_SAFETY -> "Экстренная помощь по безопасности детей"
            CYBERBULLYING -> "Помощь при кибербуллинге"
            FEEDBACK -> "Отзывы и предложения"
            SUGGESTIONS -> "Предложения по улучшению"
            COMPLAINTS -> "Жалобы и проблемы"
            OTHER -> "Другие вопросы"
        }
    
    val priority: SupportPriority
        get() = when (this) {
            CRISIS, EMERGENCY, CHILD_SAFETY, CYBERBULLYING -> SupportPriority.CRITICAL
            PSYCHOLOGICAL, TECHNICAL, TROUBLESHOOTING -> SupportPriority.HIGH
            SECURITY, PARENTAL_CONTROL, CHILD_PROTECTION, ELDERLY_PROTECTION -> SupportPriority.HIGH
            VPN, NETWORK, CONNECTIVITY, DEVICE_SECURITY -> SupportPriority.MEDIUM
            PAYMENTS, SUBSCRIPTION, BILLING -> SupportPriority.MEDIUM
            SETTINGS, CONFIGURATION, PERFORMANCE -> SupportPriority.MEDIUM
            GENERAL, HELP, FEEDBACK, SUGGESTIONS -> SupportPriority.LOW
            OTHER, COMPLAINTS -> SupportPriority.LOW
        }
    
    val icon: String
        get() = when (this) {
            GENERAL -> "ic_help_outline"
            SECURITY -> "ic_security"
            FAMILY -> "ic_family"
            SETTINGS -> "ic_settings"
            HELP -> "ic_help"
            TECHNICAL -> "ic_build"
            INSTALLATION -> "ic_download"
            CONFIGURATION -> "ic_tune"
            TROUBLESHOOTING -> "ic_bug_report"
            PERFORMANCE -> "ic_speed"
            PARENTAL_CONTROL -> "ic_child_care"
            CHILD_PROTECTION -> "ic_child_friendly"
            ELDERLY_PROTECTION -> "ic_elderly"
            DEVICE_SECURITY -> "ic_phone_android"
            NETWORK_SECURITY -> "ic_network_check"
            DATA_PROTECTION -> "ic_lock"
            VPN -> "ic_vpn_lock"
            NETWORK -> "ic_wifi"
            CONNECTIVITY -> "ic_signal_cellular_alt"
            FIREWALL -> "ic_security"
            PAYMENTS -> "ic_payment"
            SUBSCRIPTION -> "ic_repeat"
            BILLING -> "ic_receipt"
            REFUNDS -> "ic_undo"
            CRISIS -> "ic_warning"
            EMERGENCY -> "ic_emergency"
            PSYCHOLOGICAL -> "ic_favorite"
            CHILD_SAFETY -> "ic_child_care"
            CYBERBULLYING -> "ic_report"
            FEEDBACK -> "ic_feedback"
            SUGGESTIONS -> "ic_lightbulb"
            COMPLAINTS -> "ic_report_problem"
            OTHER -> "ic_more_horiz"
        }
    
    val color: String
        get() = when (this) {
            CRISIS, EMERGENCY, CHILD_SAFETY, CYBERBULLYING -> "error"
            PSYCHOLOGICAL, TECHNICAL, TROUBLESHOOTING -> "warning"
            SECURITY, PARENTAL_CONTROL, CHILD_PROTECTION, ELDERLY_PROTECTION -> "accent"
            VPN, NETWORK, CONNECTIVITY, DEVICE_SECURITY -> "primary"
            PAYMENTS, SUBSCRIPTION, BILLING -> "secondary"
            SETTINGS, CONFIGURATION, PERFORMANCE -> "info"
            GENERAL, HELP, FEEDBACK, SUGGESTIONS -> "success"
            OTHER, COMPLAINTS -> "text"
        }
}

// MARK: - Support Priority Enum
@Serializable
enum class SupportPriority(val value: String) {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high"),
    CRITICAL("critical");
    
    val displayName: String
        get() = when (this) {
            LOW -> "Низкий"
            MEDIUM -> "Средний"
            HIGH -> "Высокий"
            CRITICAL -> "Критический"
        }
    
    val description: String
        get() = when (this) {
            LOW -> "Обычный вопрос, не требует срочного решения"
            MEDIUM -> "Важный вопрос, требует внимания"
            HIGH -> "Срочный вопрос, требует быстрого решения"
            CRITICAL -> "Критическая ситуация, требует немедленного решения"
        }
    
    val responseTimeMillis: Long
        get() = when (this) {
            LOW -> 24 * 60 * 60 * 1000L // 24 часа
            MEDIUM -> 4 * 60 * 60 * 1000L // 4 часа
            HIGH -> 1 * 60 * 60 * 1000L // 1 час
            CRITICAL -> 15 * 60 * 1000L // 15 минут
        }
    
    val color: String
        get() = when (this) {
            LOW -> "success"
            MEDIUM -> "info"
            HIGH -> "warning"
            CRITICAL -> "error"
        }
}

// MARK: - Crisis Type Enum
@Serializable
enum class CrisisType(val value: String) {
    CHILD_SAFETY("child_safety"),
    CYBERBULLYING("cyberbullying"),
    PSYCHOLOGICAL("psychological"),
    EMERGENCY("emergency"),
    SECURITY("security"),
    OTHER("other");
    
    val displayName: String
        get() = when (this) {
            CHILD_SAFETY -> "Безопасность детей"
            CYBERBULLYING -> "Кибербуллинг"
            PSYCHOLOGICAL -> "Психологическая помощь"
            EMERGENCY -> "Экстренная ситуация"
            SECURITY -> "Безопасность"
            OTHER -> "Другое"
        }
    
    val description: String
        get() = when (this) {
            CHILD_SAFETY -> "Критическая ситуация с безопасностью ребенка"
            CYBERBULLYING -> "Случай кибербуллинга или травли"
            PSYCHOLOGICAL -> "Психологическая кризисная ситуация"
            EMERGENCY -> "Экстренная ситуация, требующая немедленного вмешательства"
            SECURITY -> "Критическая угроза безопасности"
            OTHER -> "Другая кризисная ситуация"
        }
    
    val urgencyLevel: Int
        get() = when (this) {
            CHILD_SAFETY, EMERGENCY -> 1
            CYBERBULLYING, SECURITY -> 2
            PSYCHOLOGICAL -> 3
            OTHER -> 4
        }
}

